bpf/include/bpf/helpers.h

```diff
@@ -84,4 +84,15 @@ static int BPF_FUNC(sock_hash_update, struct bpf_sock_ops *skops, void *map,
 static int BPF_FUNC(msg_redirect_hash, struct sk_msg_md *md, void *map,
 		    void *key, __u64 flags);
 
+/* Socket lookup helpers */
+/* sk_lookup_* apply to various contexts, including socket, sk_buff, etc.
+ * Hence, use void pointer type.
+ */
```

Comment:
I'd just remove the comment `apply to various contexts` since I think it's obvious enough from the used code.

Reply:
Well, the function is currently not used in any other context, but I'm fine either way.

Reply:
sk_lookup_* is used for BPF TProxy from skb context.

```
$ git grep sk_lookup
bpf/include/bpf/helpers_skb.h:static struct bpf_sock *BPF_FUNC(sk_lookup_udp, struct __sk_buff *skb,
bpf/include/linux/bpf.h: * struct bpf_sock *bpf_sk_lookup_tcp(void *ctx, struct bpf_sock_tuple *tuple, u32 tuple_size, u64 netns, u64 flags)
bpf/include/linux/bpf.h: * struct bpf_sock *bpf_sk_lookup_udp(void *ctx, struct bpf_sock_tuple *tuple, u32 tuple_size, u64 netns, u64 flags)
bpf/include/linux/bpf.h: *              **bpf_sk_lookup_xxx**\ ().
bpf/include/linux/bpf.h: *              This function is identical to **bpf_sk_lookup_tcp**\ (), except
bpf/include/linux/bpf.h: *              returned from **bpf_sk_lookup_xxx**\ (),
bpf/include/linux/bpf.h:        FN(sk_lookup_tcp),              \
bpf/include/linux/bpf.h:        FN(sk_lookup_udp),              \
bpf/lib/proxy.h:        sk = sk_lookup_udp(ctx, tuple, len, BPF_F_CURRENT_NETNS, 0);
```